mongodbatlas/federation_settings.go

```diff
@@ -0,0 +1,95 @@
+// Copyright 2022 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mongodbatlas
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+)
+
+const federationSettingsBasePath = "api/atlas/v1.0/orgs/%s/federationSettings"
+const federationSettingsDeleteBasePath = "api/atlas/v1.0/federationSettings"
+
+// FederatedSettingsService is an interface for working with the Federation Settings
+// endpoints of the MongoDB Atlas API.
+// See more: https://www.mongodb.com/docs/atlas/reference/api/federation-configuration/
+type FederatedSettingsService interface {
+	Get(context.Context, string) (*FederatedSettings, *Response, error)
+	Delete(context.Context, string) (*Response, error)
+}
```

Comment:
Big change, we organize the client the same way we orgnaize the docs
From [README](https://github.com/mongodb/go-client-mongodb-atlas#usage)
>The services of a client divide the API into logical chunks and correspond to the structure of the Atlas API documentation at https://docs.atlas.mongodb.com/api/.

This means this should be 
```suggestion
type FederatedAuthenticationConfigurationService interface {
	Get(context.Context, string) (*FederatedSettings, *Response, error)
	ListConnectedOrgs(...)
	GetOneConnectedOrg(...)
	UpdateConnectedOrg(...)
	RemoveConnectedOrg(...)
	ListRoleMapping(...)
	CreateRoleMapping(...)
	// And the rest following the docs
	Delete(context.Context, string) (*Response, error)
}
```

Reply:
you can keep the different files for the different sections if you want but there should be a single service all the files reference